microsoft/testsuites/cvm/cvm_attestation.py

```diff
@@ -61,11 +65,14 @@ def verify_azure_cvm_attestation_report(
         result: TestResult,
         variables: Dict[str, Any],
     ) -> None:
-        node.tools[AzureCVMAttestationTests].run_cvm_attestation(
-            result,
-            environment,
-            log_path,
-        )
+        if isinstance(node.os, Ubuntu):
+            node.tools[AzureCVMAttestationTests].run_cvm_attestation(
+                result,
+                environment,
+                log_path,
+            )
+        elif isinstance(node.os, CBLMariner):
+            assert_that(node.tools[SnpGuest].run_cvm_attestation()).is_true()
```

Comment:
Do we not want/need to use similar parameters to the existing test?

Reply:
imo since this is the actual test case definition. Assertion should be done here rather than in the tool

Reply:
If all cases need a success result, you can assert in the tool. And please always add `describe_as` to provide an error message in case it happens. So it's easy to know the error in the test results, instead of digging into the code.

Reply:
Yup. Each function in the tool already has an assert for exit code = 0, and an error message in case of failure. The assert here is more or less to make it clear what we're looking for.

Reply:
1. This line can be `node.tools[SnpGuest].run_cvm_attestation())`. It should be enough, if the method have enough assertion already.
2. In case it needs to check the return code, add `described_as` to provide error message for this `assert_that`. Please have a try to understand the behaviors of assert_that.